### PR TITLE
Add `pre-commit` hook

### PR DESCRIPTION
**Pull request by trungleduc**
## References

This PR adds a `pre-commit` hook to `Voila` to format `python` and `typescript` code.

### CODE REVIEW

**Comment by martinRenou** [comment]
Do we need to enable a bot for this?

File: package.json
Comment (jtpio) [inline package.json:1]: Should we remove the husky / lint-staged hooks?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  - repo: https://github.com/PyCQA/autoflake
+    rev: v2.0.1
+    hooks:
+      - id: autoflake
+        # args ref: https://github.com/PyCQA/autoflake#advanced-usage
+        args:
+          - --in-place
+
+  # Autoformat: Python code
+  - repo: https://github.com/pycqa/isort
```
Comment (jtpio) [inline .pre-commit-config.yaml:36]: How about adopting `ruff` directly?

Comment (jtpio) [reply .pre-commit-config.yaml:36]: A good inspiration for this would be the jupyterlab or notebook repos.

Comment (trungleduc) [reply .pre-commit-config.yaml]: Indeed, the config from `Jupyterlab` is much cleaner

File: package.json
Comment (trungleduc) [reply package.json:1]: Removed

**Comment by trungleduc** [comment]
> Do we need to enable a bot for this?

Nope, it is handled by an external service.

File: .flake8 [hunk 2]
```diff
@@ -1,3 +1,5 @@
 [flake8]
 max-line-length=200
-
+ignore = E, C, W, D
```
Comment (jtpio) [inline .flake8:3]: Do we still need the flake8 file with `ruff`?

Comment (trungleduc) [reply .flake8]: Thanks @jtpio for catching it!

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -22,47 +21,45 @@ jobs:
...
+        run: |
+          VOILA_TEST_XEUS_CLING=1 py.test tests/ --async-test-timeout=240 --reruns 2 --reruns-delay 1
+          voila --help  # Making sure we can run `voila --help`
+          # tests if voila sends a 'heartbeat' to avoid proxies from closing an apparently stale connection
+          # Note that wget is the only easily available software that has a read-timeout
+          voila tests/notebooks/sleep10seconds.ipynb --port=8878 --VoilaConfiguration.http_keep_alive_timeout=2 &
+          sleep 2
+          wget --read-timeout=5 --tries=1 http://localhost:8878
+
+      - name: Flake8
```
Comment (jtpio) [inline .github/workflows/main.yml:57]: This step could be removed as well if `ruff` already runs on CI.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -22,47 +21,45 @@ jobs:
...
+      - name: Setup mamba
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniforge-variant: Mambaforge
+          channels: conda-forge
 
-  test-osx:
+      - name: Create the conda environment
+        shell: bash -l {0}
+        run: mamba install -q python=${{ matrix.python_version }} pip jupyterlab_pygments==0.1.0 pytest-cov pytest-rerunfailures nodejs yarn=1 flake8 ipywidgets matplotlib xeus-cling openssl=1.1.1l "traitlets>=5.0.3,<6"
```
Comment (jtpio) [inline .github/workflows/main.yml:34]:
```suggestion
        run: mamba install -q python=${{ matrix.python_version }} pip jupyterlab_pygments==0.1.0 pytest-cov pytest-rerunfailures nodejs yarn=1 ipywidgets matplotlib xeus-cling openssl=1.1.1l "traitlets>=5.0.3,<6"
```

**Pull request merged**